distribution/docs/src/main/resources/_contents/_security-framework/security-framework-intro-contents.adoc

```diff
@@ -0,0 +1,169 @@
+AuthenticationToken and `org.apache.cxf.ws.security.tokenstore.SecurityToken` objects.
+
+|===
+
+==== Authentication Tokens
+
+`org.apache.shiro.authc.AuthenticationToken`
+
+Authentication Tokens are used to verify authentication of a user when creating a subject.
+A common use-case is when a user is logging directly in to the ${branding} framework.
+
+[cols="1m,2" options="header"]
+|===
+
+|Classname
+|Description
+
+|${ddf-branding-lowercase}.security.service.impl.cas.CasAuthenticationToken
+|This Authentication Token is used for authenticating a user that has logged in with CAS.
+It takes in a proxy ticket which can be validated on the CAS server.
+
+|===
+
+==== Realms
+
+${branding} uses Realms for Authentication and Authorization.
+
+===== Authenticating Realms
+
+`org.apache.shiro.realm.AuthenticatingRealm`
+
+Authenticating Realms are used to authenticate an incoming authentication token and create a Subject on successfully authentication.
+
+.Implementations of Authenticating Realms in ${branding}:
+[cols="1m,2" options="header"]
+|===
+
+|Classname
+|Description
+
+|${ddf-branding-lowercase}.security.realm.sts.StsRealm
+|This realm delegates authentication to the Secure Token Service (STS). It creates a `RequestSecurityToken` message from the incoming Authentication Token and converts a successful STS response into a Subject.
+
+|===
+
+===== Authorizing Realms
+
+`org.apache.shiro.realm.AuthorizingRealm`
+
+Authorizing Realms are used to perform authorization on the current Subject.
+These are used when performing both Service AuthZ and Filtering.
+They are passed in the `AuthorizationInfo` of the Subject along with the Permissions of the object wanting to be accessed. The response from these realms is a true (if the Subject has permission to access) or false (if the Subject does not).
+
+.Other implementations of the Security API within ${branding}
+[cols="1m,2" options="header"]
+|===
+
+|Classname
+|Description
+
+|org.codice.${ddf-branding-lowercase}.platform.filter.delegate.DelegateServletFilter
+|The `DelegateServletFilter` detects any servlet filters that have been exposed as OSGi services and places them in-order in front of any servlet or web application running on the container.
+
+|org.codice.${ddf-branding-lowercase}.security.filter.websso.WebSSOFilter
+|This filter serves as the main security filter that works in conjunction with a number of handlers to protect a variety of contexts, each using different authentication schemes and policies.
+
+|org.codice.${ddf-branding-lowercase}.security.handler.saml.SAMLAssertionHandler
+|This handler is executed by the WebSSOFilter for any contexts configured to use it.
+This handler should always come first when configured in the Web Context Policy Manager, as it provides a caching capability to web contexts that use it.
+The handler will first check for the existence of a cookie named "org.codice.websso.saml.token" to extract a Base64 + deflate SAML assertion from the request.
```

Comment:
The handler will first check for the existence of an HTTP Authorization header of type SAML, whose value is a Base64 + deflate SAML assertion. If that is not found, then the handler will check for the existence of the deprecated "org.codice.websso.saml.token" cookie with the same value. Failing that, it will check for a JSESSIONID cookie to use as a reference to a cached assertion.